src/flownet/ert/_create_ert_setup.py

```diff
@@ -23,6 +24,42 @@
 _MODULE_FOLDER = pathlib.Path(os.path.dirname(os.path.realpath(__file__)))
 
 
+def resample_schedule_dates(schedule: Schedule, resampling: Optional[str]) -> List:
+    """
+    Resampling dates based on requested frequency without introducing interpolated dates,
+    i.e., keeps nearest existing date
+
+    Args:
+        schedule: FlowNet Schedule instance to create observations from.
+        resampling: Pandas resampling frequency (str)
+
+    Returns:
+        dates: list of resampled dates
+
+    """
+    df_dates_schedule = pd.to_datetime(schedule.get_dates())
+
+    if resampling:
+        df_dates_resampled = pd.date_range(
+            schedule.get_dates()[0],
+            schedule.get_dates()[-1],
+            freq=resampling,
+        )
+        df_schedule = pd.DataFrame(
+            data=range(len(df_dates_schedule)), index=df_dates_schedule
+        )
+        idx = np.zeros(len(df_dates_resampled), dtype=int)
+        for i, k in enumerate(df_dates_resampled):
+            idx[i] = df_schedule.index.get_loc(k, method="nearest")
+        dates = [
+            d.date() for d in df_schedule.iloc[np.unique(idx)].index.to_pydatetime()
+        ]
```

Comment:
Should you check if the first resampled date is equal to the original first date; if so: remove it!?

Reply:
this is not needed, because it will only loop through the range of dates starting at the second date